src/backend/distributed/planner/intermediate_result_pruning.c

```diff
@@ -206,17 +220,23 @@ MakeIntermediateResultHTAB()
  * may need to access subplan results.
  */
 List *
-FindAllWorkerNodesUsingSubplan(HTAB *intermediateResultsHash,
+FindAllWorkerNodesUsingSubplan(IntermediateResultsHashEntry *entry,
 							   char *resultId)
 {
 	List *workerNodeList = NIL;
-	IntermediateResultsHashEntry *entry =
-		SearchIntermediateResult(intermediateResultsHash, resultId);
 
 	ListCell *nodeIdCell = NULL;
 	foreach(nodeIdCell, entry->nodeIdList)
 	{
-		WorkerNode *workerNode = LookupNodeByNodeId(lfirst_int(nodeIdCell));
+		uint32 nodeId = lfirst_int(nodeIdCell);
+		WorkerNode *workerNode = LookupNodeByNodeId(nodeId);
+
+		if (workerNode == NULL)
+		{
+			elog(DEBUG4, "Failed Node Lookup with Id %d, write to local file", nodeId);
+			entry->writeLocalFile = true;
```

Comment:
how can this happen?

Reply:
If we have a dummy placement that points to the coordinator that does not have a valid record in `pg_dist_node` I assumed that it should be written to a local file.

I will introduce a new placeholder such as `INVALID_NODE_ID=0` that will be used for this purpose. Does this sound good?

Reply:
When creating a dummy placement, the `nodeId` field will now be `DUMMY_NODE_ID = -1` and before doing a lookup, I will check if this instance is indeed a dummy placement.